monitoring.go

```diff
@@ -318,40 +318,46 @@ func (sc *snowflakeConn) rowsForRunningQuery(
 
 // Wait for query to complete from a query id from /queries/<qid>/result endpoint.
 func (sc *snowflakeConn) blockOnRunningQuery(
-    ctx context.Context, qid string) error {
-    resultPath := fmt.Sprintf(urlQueriesResultFmt, qid)
-    resp, err := sc.waitForCompletedQueryResultResp(ctx, resultPath)
-    if err != nil {
-        logger.WithContext(ctx).Errorf("error: %v", err)
-        if resp != nil {
-            code, err := strconv.Atoi(resp.Code)
-            if err != nil {
-                return err
-            }
-            return (&SnowflakeError{
-                Number:   code,
-                SQLState: resp.Data.SQLState,
-                Message:  err.Error(),
-                QueryID:  resp.Data.QueryID,
-            }).exceptionTelemetry(sc)
-        }
-        return err
-    }
-    if !resp.Success {
-        message := resp.Message
-        code, err := strconv.Atoi(resp.Code)
-        if err != nil {
-            code = ErrQueryStatus
-            message = fmt.Sprintf("%s: (failed to parse original code: %s: %s)", message, resp.Code, err.Error())
-        }
-        return (&SnowflakeError{
-            Number:   code,
-            SQLState: resp.Data.SQLState,
-            Message:  message,
-            QueryID:  resp.Data.QueryID,
-        }).exceptionTelemetry(sc)
-    }
-    return nil
+	ctx context.Context, qid string) error {
+	resultPath := fmt.Sprintf(urlQueriesResultFmt, qid)
+	resp, err := sc.waitForCompletedQueryResultResp(ctx, resultPath)
+	if err != nil {
+		logger.WithContext(ctx).Errorf("error: %v", err)
+		if resp != nil {
```

Comment:
Will `sc.waitForCompletedQueryResultResp` ever return a non-nil `resp` if there was an error? I feel like we may be able to ignore this case.

Reply:
I dont know why it would but they do this "if error and resp..." logic in `rowsForRunningQuery` which makes me worry that in some strange case FuncGet could return an error and a response... which I definitely dont like. Maybe it would be best to just say if there is an error return the error and ignore the response to keep things consistent

Reply:
ok I think I will keep it in here for now but if you think it would be better to remove it I can make a follow up pr!

Reply:
ok, no worries